cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -622,6 +622,22 @@ class list_scalar : public scalar {
   {
   }
 
+  /**
+   * @brief Move existing device data into a new list scalar object
+   *
+   * @param elements The elements of the list
+   * @param is_valid Whether the value held by the scalar is valid
+   * @param stream CUDA stream used for device memory operations.
+   * @param mr Device memory resource to use for device memory allocation
+   */
+  list_scalar(cudf::column const&& elements,
+              bool is_valid                       = true,
+              rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
+              rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
+    : scalar(data_type(type_id::LIST), is_valid, stream, mr), _data(elements)
+  {
+  }
+
   /**
```

Comment:
I think it would be useful to have a constructor that behaves just like the factory function.   Something like:

```
  list_scalar(cudf::column_view const& col,
              size_type row_index,
              rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
              rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
```

Reply:
Does this mean constructing a `list_scalar` from the `row_index`th row of `col`?

Reply:
Yes

Reply:
Then why have the factory?

Reply:
We've got these other constructors.  If we have them, why not have another one that provides utility?  Or to put it another way, if we shouldn't have this one, why should we have the other ones?

Reply:
We've got these other constructors. If we have them, why not have another one that provides utility? Or to put it another way, if we shouldn't have this one, why should we have the other ones?

Reply:
Second thoughts: for other scalar types there is always a constructor that constructs the scalar from its "value". `string_scalar(std::string)`, `numeric_scalar(T value)` etc. So it makes sense to have a constructor for `list_scalar` accepting its underlying data.

For the proposed ctor - note this doesn't have to be specific to `list_scalar`, it can actually be implemented in `scalar` base class by punting everything to `get_element`:
```cpp
scalar::scalar(column_view const& col,
          size_type row_index,
          rmm::cuda_stream_view stream,
          rmm::mr::device_memory_resource* mr)
          : scalar(std::move(*detail::get_element(col, row_index, stream, mr).release()))
{}
```
But then this is just another level of indirection.